Detectors/TPC/calibration/include/TPCCalibration/CalibPadGainTracks.h

```diff
@@ -0,0 +1,521 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// @file   CalibPadGainTracks.h
+/// @author Matthias Kleiner, matthias.kleiner@cern.ch
+///
+
+#ifndef AliceO2_TPC_CalibPadGainTracks_H
+#define AliceO2_TPC_CalibPadGainTracks_H
+
+//o2 includes
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsTPC/ClusterNative.h"
+#include "TPCBase/CalDet.h"
+#include "TPCBase/ROC.h"
+#include "TPCBase/PadPos.h"
+#include "TPCBase/Mapper.h"
+
+// for FastHistogram class
+#include "MathUtils/MathBase.h"
+
+//root includes
+#include "TFile.h"
+#include "TTree.h"
+
+#include <vector>
+#include <iostream>
+#include <tuple>
+
+namespace o2
+{
+namespace tpc
+{
+
+/// \brief templated 1D-histogram class.
+///
+/// This is a templated histogram class without any ROOT includes.
+///
+/// How to use:
+/// e.g.: create an array with 7 bins + underflow bin + overflow bins
+/// bin range is from xmin=0.4f to xmax=34.6f
+/// the values which will be filled are of the type float
+/// o2::tpc::FastHisto<float> histo(7,0.4f,34.6f,true,true);
+/// histo.fill(4.5);
+/// histo.fill(20.2,3);
+/// histo.fill(35);
+/// histo.print()
+/// float mean = histo.getStatisticsData(0,1).mCOG;
+/// float TruncatedMean = histo.getStatisticsData(0.05,.6).mCOG;
```

Comment:
It might be better to put the FastHisto class in an own file. Otherwise readability is not so good.

Reply:
Okay, I moved the FastHisto class to FastHisto.h to the include/TPCCalibration folder for now.